### DISCUSSION TITLE
0.10.0-rc1 has been released!

### DISCUSSION THREAD

**Discussion by mihe** [godot-jolt/godot-jolt#667] [Announcements]
This is the first **release candidate** for version 0.10.0, and assuming no major bugs/regressions are found is likely to become 0.10.0-stable.

This release supports [**Godot 4.1**](https://github.com/godotengine/godot/releases/tag/4.1.3-stable).

## Changelog

These are the notable changes that have been made since [0.9.0-stable](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.9.0-stable) was released. You can also find a list of all the commits [here](https://github.com/godot-jolt/godot-jolt/compare/v0.9.0-stable...v0.10.0-rc1).

Any breaking changes are denoted with ⚠️.

### Changed

- ⚠️ Changed so that ray-casts using the `hit_from_inside` parameter report a zero normal when hitting a convex shape from inside, to match Godot Physics.
- Changed the `space_get_contacts` method of `PhysicsServer3D` (and thus also the "Visible Collision Shapes" debug rendering) to no longer include contacts generated by overlaps with `Area3D`.

### Added

- Added support for Android (ARM64, ARM32, x86-64 and x86).
- Added support for iOS.

### Fixed

- Fixed issue where an error saying `Parameter "body" is null` would be emitted after freeing certain bodies while they were in contact with a `CharacterBody3D`.

<hr /><em>This discussion was created from the release <a href='https://github.com/godot-jolt/godot-jolt/releases/tag/v0.10.0-rc1'>0.10.0-rc1</a>.</em>

**Comment by mihe** [comment]
To anyone stumbling onto this release from somewhere other than #353 or #354: I'd be very curious to hear if/how this release works on your Android/iOS device, as I only have one of each to test with myself. I'm especially curious about more specialized Android devices like the Quest VR headsets.

**Reply by phelioz** [reply]
Great work as always!
I tested the VR game I have been developing with Godot Jolt on my Quest 2, and it just works 😃. However, I did encounter some heavy stutters at times, but I have mainly been developing for PCVR, so it's probably just my project that needs some tweaking.

Of course, I will post an issue if I encounter something related to Godot Jolt when running on the Quest 2. I will get a Quest 3 in a couple of days, so I will test that as well.

**Reply by mihe** [reply]
That's a relief. Thank you for trying it out.

**Reply by Ughuuu** [reply]
Also tested on iPhone 13 mini and works. Also ran the iOS version on Mac(with m1 so it can run iOS apps). Works perfect on both.

**Reply by RobvandenBerg** [reply]
I was feeling adventurous so I tried importing it in Godot 4.**2** (instead of 4.1), exported the project to my Meta Quest 3, and... Worked like a charm! Great job!

**Reply by mihe** [reply]
@RobvandenBerg Are you noticing any stutters or generally worse performance compared to Godot Physics when running it on your Quest 3? I received a report from elsewhere that performance was worse than Godot Physics when running it on a PICO 4, although I didn't get any specifics about the exact scenario, so now I'm wondering if maybe what @phelioz mentioned wasn't a coincidence.

I'll see about doing some profiling on my phone in the meantime.

EDIT: I'm seeing much better performance with just some very basic stress tests, so I'm guessing it was probably a manifestation of #626.

**Reply by RobvandenBerg** [reply]
@mihe If I compare spawning a lot of cubes on JoltPhysics3D to the Godot Physics engine, then Jolt indeed seems to have worse performance on my Quest 3.

**Reply by mihe** [reply]
It appears the problem is as simple as Jolt not running with any optimizations enabled whatsoever when building its `Distribution` configuration on Android, whereas its `Release` configuration does, which explains why it ran like a dream for me when I ran my own development builds, but once I ran the actual distributed 0.10.0-rc1 binaries then it turned into a slideshow.

That was a dumb mistake on my part. Now I just need to figure out how to fix this.

**Reply by mihe** [reply]
New (optimized) release here: [0.10.0-rc2](https://github.com/godot-jolt/godot-jolt/releases/tag/v0.10.0-rc2)
